packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -63,6 +63,10 @@ import type {
 import type { VM } from './vm';
 import type { RendererAPI } from './renderer';
 
+type Classes = Omit<Set<string>, 'add'>;
+
+const EMPTY_SET: Classes = new Set<string>();
```

Comment:
We don't want to accidentally do `EMPTY_SET.add(something)`, so this is just a type hack to warn ourselves.

Reply:
Let's add a short comment explaining this optimization.